buffer.go

```diff
@@ -36,61 +44,74 @@ func (b *Buffer) HandleMeasures(time time.Time, measures ...Measure) {
 		return
 	}
 
-	size := b.BufferSize
-	if size == 0 {
-		size = 1024
-	}
-
-	chunk := acquireChunk()
-	chunk.reset()
-	chunk.append(b.Serializer, time, measures...)
-
-	// Chunks that are already larger than the maximum buffer size are directly
-	// passed to the serializer, no need to aggregate them in a larger buffer.
-	if chunk.len() >= size {
-		chunk.writeTo(b.Serializer)
-	} else {
-		for {
-			var flush *buffer
+	size := b.bufferSize()
+	b.init(size)
 
-			b.mutex.Lock()
+	for {
+		buffer := b.acquireBuffer()
+		length := buffer.len()
+		buffer.append(b.Serializer, time, measures...)
 
-			if b.buffer == nil {
-				b.buffer = acquireBuffer()
-				b.buffer.reset(size)
-			}
+		if buffer.len() < size {
+			buffer.release()
+			break
+		}
 
-			if (b.buffer.len() + chunk.len()) > b.buffer.cap() {
-				b.buffer, flush = nil, b.buffer
-			} else {
-				b.buffer.append(chunk)
-			}
+		if length != 0 {
+			buffer.trimTo(length)
+		}
 
-			b.mutex.Unlock()
+		buffer.writeTo(b.Serializer)
+		buffer.reset()
+		buffer.release()
+	}
+}
 
-			if flush == nil {
-				break
-			}
+// Flush satisfies the Flusher interface.
+func (b *Buffer) Flush() {
+	b.init(b.bufferSize())
```

Comment:
Is this meant to be here?  `init` looks like it zeros buffers to me.

Maybe that's what you want, but it doesn't sound like a `Flush` type to me.  However, I don't know how this is used as a flusher.

Reply:
I see, init is a `once.Do` and we don't want to flush before they're initialized.

A bit confusing to the reader still.  Wonder if there's a cleaner way.  It's also confusing going between `Buffer` and `buffer` types in this file when they share method names.

Reply:
How about renaming the method initOnce, or prepare?

Reply:
I think I was more confused by the two init methods more so than the name itself.  I wasn't in my editor and went to the other function when reviewing in the browser.

It's just a nit mostly.